### PR TITLE
adding mirror credential support

### PR DESCRIPTION
**Pull request by dgraeber**
*Issue #, if available:*
#163 

*Description of changes:*
Adding support for credentials on mirrors for pypi

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws_codeseeder/resources/pypi_mirror_support.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        key = secret_name.split("::")[1] if "::" in secret_name else "pypi"
+        creds = get_secret(secret_name=secret_name_key)
+        if key in creds.keys():
+            username = creds[key]["username"] if creds[key].get("username") else None
+            password = creds[key]["password"] if creds[key].get("password") else None
+
+    secured_url, obfusctated_url = create_url(url, username, password)
+    logger.info("Calling pip config with %s", obfusctated_url)
+    print(f"pip config being set for : {obfusctated_url}")
+    subprocess.call(f"pip config set global.index-url {secured_url}", shell=True)
```
Comment (kukushking) [inline aws_codeseeder/resources/pypi_mirror_support.py:63]: How about this instead:
`subprocess.call(["pip", "config", "set", "global.index-url", secured_url])`

File: aws_codeseeder/resources/pypi_mirror_support.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+from typing import Dict, Optional, Tuple, cast
+
+import boto3
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+# os.environ["AWS_CODESEEDER_MIRROR_SECRET"]="derek/idf-mirror-credentials::artifactory"
```
Comment (srinivasreddych) [inline aws_codeseeder/resources/pypi_mirror_support.py:17]: can you delete this comment?

Comment (dgraeber) [reply aws_codeseeder/resources/pypi_mirror_support.py:17]: yes

File: aws_codeseeder/resources/pypi_mirror_support.py (hunk 1 above)
Comment (dgraeber) [reply aws_codeseeder/resources/pypi_mirror_support.py]: yes

**Review by srinivasreddych — APPROVED** [review]

**Review by kukushking — COMMENTED** [review]
Looks good, thanks!

**Pull request merged**